test/ossl_store_test.c

```diff
@@ -51,6 +51,36 @@ static int test_store_search_by_key_fingerprint_fail(void)
     return ret;
 }
 
+// This test verifies that calling OSSL_STORE_ATTACH does not set an
+// "unregistered scheme" error when called.
+//
+// This test needs to force the code path in OSSL_STORE_attach to call
+// ossl_store_get0_loader_int which is done by undefining
+// OPENSSL_NO_DEPRECATED_3_0.
+static int test_store_attach_unregistered_scheme(void)
+{
+#undef OPENSSL_NO_DEPRECATED_3_0
```

Comment:
This isn't a good idea.  What if it's not defined and gets defined at the end?

The build time configuration options set this define or not.

Reply:
Ah right, good point. Let me take another stab at this. Thanks